libs/base/Data/String.idr

```diff
@@ -0,0 +1,49 @@
+module Data.String
+
+||| Convert a positive number string to a Num.
+|||
+||| ```idris example
+||| parsePositive "123"
```

Comment:
The syntax for examples doesn't work quite that way. Each example needs to be in its own block to have it render properly when viewing docs:
`````
||| ```idris example
||| parsePositive "123"
||| ```
||| ```idris example
||| parsePositive {a=Int} " +123"
||| ```
`````